### PR TITLE
Improved error wrapping and add support for object alias

### LINKED ISSUE #51: Allow to specify an Alias for the file name
In the original [docs/Concept.md](https://github.com/Azure/kubernetes-keyvault-flexvol/blob/master/docs/concept.md) there is an optional `Alias` parameter, which is not implemented, but would indeed be useful.

Note that this would needs to consider multiple secret compatibility https://github.com/Azure/kubernetes-keyvault-flexvol/pull/49

### LINKED ISSUE #66: Support .NET configuration conventions for structured data
The .NET convention for structured keys is using underscores where json or other structured format is not available, like environment variable names and, in this case, file names.

For some reason, Azure Key Vault doesn't allow underscores in key names, so dashes are used instead. And the logical thing is to use the `KeyPerFileConfigurationProvider` and call `config.AddKeyPerFile("/kv")` to load all values. But this provides doesn't handle double dashes.

It would be stupid to write yet another configuration provider to handle this, and when the convention is established, the FlexVolume provider should replace double dashes with double underscores. Or provide a variable to specify the delimiter.

### PR DESCRIPTION
**Pull request by jon-walton**
Hi,

I needed object aliases so this PR adds that. It works the same as described in concept.md. It also adds some improvements to the error wrapping that helped debugging my deployment

I did initially set the k8s config to be `keyvaultobjectaliases` until I saw concept.md, let me know if you want it renamed.

fixes #51 
fixes #66

### CODE REVIEW

**Comment by ritazh** [comment]
Thanks for your contribution @jon-walton 🎉 Will review this PR shortly.

File: azurekeyvault-flexvolume/keyvaultFlexvolumeAdapter.go [hunk 1]
```diff
@@ -52,11 +52,17 @@ func (adapter *KeyvaultFlexvolumeAdapter) Run() error {
...
 	objectNames := strings.Split(options.vaultObjectNames, objectsSep)
+	objectAliases := strings.Split(options.vaultObjectAliases, objectsSep)
 	objectVersions := strings.Split(options.vaultObjectVersions, objectsSep)
 
 	for i := range objectNames {
 		objectType := objectTypes[i]
 		objectName := objectNames[i]
+		// default to the objectName and override if aliases are available
+		fileName := path.Join(options.dir, objectNames[i])
+		if len(objectAliases) == len(objectNames) {
```
Comment (ritazh) [inline azurekeyvault-flexvolume/keyvaultFlexvolumeAdapter.go]:
```suggestion
                 if options.vaultObjectAliases != "" && len(objectAliases) == len(objectNames) {
```

Comment (ritazh) [reply azurekeyvault-flexvolume/keyvaultFlexvolumeAdapter.go]: If `options.vaultObjectAliases = ""` then `len(objectAliases) = 1`

File: azurekeyvault-flexvolume/keyvaultFlexvolumeAdapter.go [hunk 2]
```diff
@@ -52,11 +52,17 @@ func (adapter *KeyvaultFlexvolumeAdapter) Run() error {
...
+			fileName = path.Join(options.dir, objectAliases[i])
+		}
 		// objectVersions are optional so we take as much as we can
 		objectVersion := ""
 		if len(objectVersions) == len(objectNames) {
```
Comment (ritazh) [inline azurekeyvault-flexvolume/keyvaultFlexvolumeAdapter.go]:
```suggestion
		if options.vaultObjectVersions != "" && if len(objectVersions) == len(objectNames) {
```

**Comment by ritazh** [comment]
> I did initially set the k8s config to be keyvaultobjectaliases until I saw concept.md, let me know if you want it renamed.

I think `keyvaultobjectaliases` is better as it is more consistent with other configs for each key vault object. Let's update this as concept.md is a bit out of date. Thanks @jon-walton!

**Comment by jon-walton** [comment]
thanks @ritazh , I've pushed a new commit containing your feedback

**Pull request merged**